[DOCS-1697] Minimal Operator doc update for Bitnami changes

mdlinville opened the pull request:
Description
-----------
Updates Operator airgapped install instructions to use the `bitnamilegacy` image for Redis

✅ technical review
✅ final PM review
Ready for peer review

Related issues
-----------
- Partially fixes DOCS-1697


<!-- preview-links-comment -->
📄 **[View preview links for changed pages](https://github.com/wandb/docs/pull/1562#issuecomment-3215803467)**

mdlinville committed:
[DOCS-1697] Minimal Operator doc update for Bitnami changes

mdlinville commented on content/en/guides/hosting/hosting-options/self-managed/kubernetes-operator/operator-airgapped.md:
```diff
@@ -26,20 +26,26 @@ Before starting, make sure your environment meets the following requirements:
+* [`docker.io/bitnamilegacy/redis`](https://hub.docker.com/r/bitnamilegacy/redis).
+
+  {{% alert %}}
+  W&B is currently testing the [`docker.io//bitnamisecure/redis`](https://app-catalog.vmware.com/bitnami/apps/8b68d0f6-8fb6-4a99-accd-83e4f343522f) image, which is not yet supported.
```
```suggestion
  W&B is currently testing the [`docker.io/bitnamisecure/redis`](https://app-catalog.vmware.com/bitnami/apps/8b68d0f6-8fb6-4a99-accd-83e4f343522f) image, which is not yet supported.
```

mdlinville committed:
Update content/en/guides/hosting/hosting-options/self-managed/kubernetes-operator/operator-airgapped.md

mdlinville committed:
Merge branch 'main' into DOCS-1697_take2

BrenSea-wb commented on content/en/guides/hosting/hosting-options/self-managed/kubernetes-operator/operator-airgapped.md:
```diff
@@ -26,20 +26,26 @@ Before starting, make sure your environment meets the following requirements:
+You are responsible for tracking the Operator's requirements and for checking for and downloading image upgrades regularly.
 
-You are responsible for tracking the Operator's requirements and for checking for and downloading image upgrades, either by [using WSM]({{< relref "#list-images-and-their-versions" >}}) or by using your organization's own processes.
+In production, W&B recommends against using the built-in MySQL and Redis services, which are provided for convenience during testing and early development. Instead, W&B recommends that your your deployment uses external MySQL and Redis deployments that meet your organization's requirements for high availability, disaster recovery, security, and performance.
```
typo "`**your your** `deployment uses external MySQL"

mdlinville replied:
Thanks, I ended up removing this exact change in favor of a different rewrite, which made this moot.

BrenSea-wb commented on content/en/guides/hosting/hosting-options/self-managed/kubernetes-operator/operator-airgapped.md:
```diff
@@ -26,20 +26,26 @@ Before starting, make sure your environment meets the following requirements:
+* [`docker.io/bitnamilegacy/redis`](https://hub.docker.com/r/bitnamilegacy/redis).
+
+  {{% alert %}}
+  W&B is currently testing the [`docker.io/bitnamisecure/redis`](https://app-catalog.vmware.com/bitnami/apps/8b68d0f6-8fb6-4a99-accd-83e4f343522f) image, which is not yet supported.
```
Not sure if this last statement is worth making customer facing. By the time customers see it, it'll be live and ready (not in testing anymore0

mdlinville replied:
I removed it. My understanding was that the `bitnamilegacy` package is what we can recommend right now, and that the `bitnamisecure` testing will take a little longer, so I will avoid mentioning it.

BrenSea-wb replied:
We aren't testing the bitnamisecure image at all. W&B and CW will support legacy. And eventually move off to a different solution. We be supporting the new image. There's no current tetsing.

BrenSea-wb commented on content/en/guides/hosting/hosting-options/self-managed/kubernetes-operator/operator-airgapped.md:
```diff
@@ -26,20 +26,26 @@ Before starting, make sure your environment meets the following requirements:
+You are responsible for tracking the Operator's requirements and for checking for and downloading image upgrades regularly.
 
-You are responsible for tracking the Operator's requirements and for checking for and downloading image upgrades, either by [using WSM]({{< relref "#list-images-and-their-versions" >}}) or by using your organization's own processes.
+In production, W&B recommends against using the built-in MySQL and Redis services, which are provided for convenience during testing and early development. Instead, W&B recommends that your your deployment uses external MySQL and Redis deployments that meet your organization's requirements for high availability, disaster recovery, security, and performance.
```
Here is the Bitnami customer communication that we wrote: https://docs.google.com/document/d/1QpiNBqVod9vf5kIiJ9beLOlabFDowly7GrgoJFejV9I/edit?tab=t.0

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1697_take2

mdlinville committed:
Adjustments

- Subdivide the images into W&B ones and dependencies
- Adjust the Redis details to align with https://github.com/wandb/docs/pull/1566
- Remove detail about bitnamisecure
- Borrow dependency descriptions from Reference Architecture

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1697_take2

mdlinville committed:
Merge remote-tracking branch 'origin/DOCS-1697_take2' into DOCS-1697_take2

mdlinville committed:
Merge branch 'main' into DOCS-1697_take2

flamarion commented on content/en/guides/hosting/hosting-options/self-managed/kubernetes-operator/operator-airgapped.md:
```diff
@@ -26,20 +26,22 @@ Before starting, make sure your environment meets the following requirements:
 
-If your organization already uses an internal container registry, you can add the images to it. Otherwise, follow the proceeding section to use a called WSM to prepare the container repository.
+* [`docker.io/bitnamilegacy/redis`](https://hub.docker.com/r/bitnamilegacy/redis): W&B depends on a single-node Redis 7.x deployment to handle job queuing and data caching used by W&B's components. For convenience during testing and development of proofs of concept, W&B Self-Managed deploys a local Redis deployment that is not appropriate for production deployments. To use the local Redis deploym ...
+* [`docker.io/otel/opentelemetry-collector-contrib`](https://hub.docker.com/r/otel/opentelemetry-collector-contrib): W&B depends on the OpenTelemetry agent to collect metrics and logs from resources at the Kubernetes layer for display in W&B.
+* [`quay.io/prometheus/prometheus`](https://quay.io/repository/prometheus/prometheus): W&B depends on Prometheus to capture metrics from various components for display in W&B.
+* [`quay.io/prometheus-operator/prometheus-config-reloader`](https://quay.io/repository/prometheus-operator/prometheus-config-reloader): A required dependency of Prometheus.
```
Just in case you believe it's relevant to make it explicit, but these two (Otel and Prometheys) are not blockers for the deployment.
I'm mentioning that because most customers will already have the Otel in their cluster, in case they deploy the W&B in a shared Kubernetes cluster.
Honestly, 80%+ of all airgapped deployments don't use it for other reasons, like a wide open access the Otel needs in the cluster.

mdlinville replied:
https://wandb.atlassian.net/browse/DOCS-1715

BrenSea-wb reviewed (dismissed):
This all looks good. Not sure if the bitnamisecure part called out in the description will show up anywhere. As long as that part is removed, this looks good.

mdlinville commented:
> This all looks good. Not sure if the bitnamisecure part called out in the description will show up anywhere. As long as that part is removed, this looks good.

It wouldn't have shown up anywhere but in the commit history, which could confuse us down the track. Fixed. Thanks.

mdlinville committed:
Merge branch 'main' into DOCS-1697_take2

mdlinville committed:
Merge branch 'main' into DOCS-1697_take2

mdlinville committed:
Merge branch 'main' into DOCS-1697_take2

ngrayluna commented on content/en/guides/hosting/hosting-options/self-managed/kubernetes-operator/operator-airgapped.md:
```diff
@@ -26,20 +26,22 @@ Before starting, make sure your environment meets the following requirements:
 ## Step 2: Prepare internal container registry
 
-Before proceeding with the deployment, you must ensure that the following container images are available in your internal container registry:
+Before proceeding with the deployment, you must ensure that the following container images are available in your internal container registry. These images are critical for the successful deployment of W&B components. W&B recommends that you either follow your organization's own processes for managing your container registry, or that you use [WSM](#install-wsm) to prepare it.
```
Double-checking the usage of "ensure". (I had several PRs were you suggested alternative verbs.)
